tests/ui/diagnostic_namespace/do_not_recommend/stacked.current.stderr

```diff
@@ -0,0 +1,25 @@
+error[E0277]: the trait bound `(): Root` is not satisfied
+  --> $DIR/stacked.rs:19:18
+   |
+LL |     needs_root::<()>();
+   |                  ^^ the trait `Other` is not implemented for `()`, which is required by `(): Root`
+   |
+note: required for `()` to implement `DontRecommend`
```

Comment:
Yeah, see, this is what I mentioned. We are recommending a do_not_recommend trait 😅

Reply:
Hmm, yes.
It's not clear for me how to actually walk the tree here, as the `ObligationCause` seems to be the `ExprBindingObligation` variant in this case (according to a `dbg!()` call in the `apply_do_not_recommend` function). That version does not seem to contain any parent tree information, as far as I understand.

Reply:
Turns out that the problem was not the loop, but the location where I called the loop. It should work now as expected.